File: Java/app/src/main/java/dev/ogabek/java/SecondActivity.java
package dev.ogabek.java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        initViews();
        getInfo();

    }

    void initViews() {
        textView = findViewById(R.id.textView);
    }

    void getInfo() {
        Me me = (Me) getIntent().getSerializableExtra("user");
        textView.setText(me.toString());
    }

}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/SecondActivity.kt
package dev.ogabek.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class SecondActivity : AppCompatActivity() {

    private lateinit var textView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        initViews()
        getInfo()

    }

    private fun initViews() {
        textView = findViewById(R.id.textView)
    }

    private fun getInfo() {
        val me = intent.getSerializableExtra("user")
        textView.text = me.toString()
    }

}